### ISSUE TITLE
Auth Routes

### ISSUE THREAD

**Issue by JackyChiu** [JackyChiu/realworld-starter-kit#3]
- [x] Login
- [ ] Login And Remeber Token
- [x] Register
- [ ] Current User
- [ ] Update User

**Comment by qwo** [comment]
Curious if these were are suppose to be swapped around.
https://github.com/JackyChiu/realworld-starter-kit/blob/master/main.go#L24-L25
Accessing users require the user to be signed in but logging in shouldn't require the token?

**Comment by adelowo** [comment]
@stanzheng nice catch, that should be swapped

**Comment by JackyChiu** [comment]
Hey, @stanzheng and @adelowo! Yep,  I definitely got that swapped up, thanks for catching that! I actually need to structure the handlers better as well since I have my `Registration` (which doesn't require auth) in  my `UsersRouter` 

If you guys have any suggestions for any parts of the design, feel free to drop me a line anytime!

**Comment by qwo** [comment]
Yeah! I'm waiting for the auth to be a bit wired up and maybe ill be able to jump in on something. thanks for spear heading this!

**Comment by guillaumemaka** [comment]
Hi I notice that the authorization header should be `Token` not `Bearer` according to the [Api Specs](https://github.com/gothinkster/realworld/tree/master/api#authentication-header)
> Authentication Header:
Authorization: Token jwt.token.here

**Comment by JackyChiu** [comment]
@guillaumemaka  Yep I'm fixing that in the `auth` package, the spec was recently changed
